### PR TITLE
Document change for DDP enabled on Windows platform

### PR DESCRIPTION
**Pull request by gunandrose4u**
Document change for DDP enabled on Windows platform

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -184,6 +184,13 @@ On MacOS
 conda install pkg-config libuv
 ```
 
+On Windows
+```bash
+# Add these packages and set libuv_ROOT environment variable if torch.distributed is needed
+conda install -y -q -c rdonnelly libuv
+set libuv_ROOT={conda active env location}\Library
```
Comment (mrshenli) [inline README.md:191]: I wonder why MacOS does not require this? MacOS requires install pkg-config instead. Can we do the same for Windows?

Comment (gunandrose4u) [reply README.md:191]: In Gloo PR https://github.com/pytorch/pytorch/pull/45392, I first tried with pkg-config, but Windows doesn't come with pkg-config, later on, base on the code review suggestion, I switched to use find_library and find_file to work with libuv_ROOT. So libuv_ROOT need to be set as a environment variable, or defined in cmake command line as -Dlibuv_ROOT.

Comment (mrshenli) [reply README.md:191]: > but Windows doesn't come with pkg-config, later on, base on the code review suggestion

Does this mean even if we do `conda install pkg-config libuv`, we still cannot find libuv on Windows automatically?

File: README.md [hunk 2]
```diff
@@ -184,6 +184,13 @@ On MacOS
 conda install pkg-config libuv
 ```
 
+On Windows
+```bash
+# Add these packages and set libuv_ROOT environment variable if torch.distributed is needed
```
Comment (mrshenli) [inline README.md:189]: Let's add one line:

```
# Distributed package support on Windows is a prototype feature and is subject to changes. 
```

Comment (mrshenli) [reply README.md:189]: @jspisak @gchanan is it OK to mention a prototype feature in master `README.md`?

File: docs/source/distributed.rst [hunk 3]
```diff
@@ -52,12 +52,19 @@ MPI supports CUDA only if the implementation used to build PyTorch supports it.
 Backends that come with PyTorch
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-PyTorch distributed currently only supports Linux. By default, the Gloo and NCCL backends
+PyTorch distributed currently only supports Linux and Windows. By default for Linux, the Gloo and NCCL backends
```
Comment (mrshenli) [inline docs/source/distributed.rst:55]: "PyTorch distributed currently only supports Linux and Windows." -> "PyTorch distributed package supports Linux (stable), MacOS (stable), and Windows (prototype)."

File: docs/source/distributed.rst [hunk 4]
```diff
@@ -52,12 +52,19 @@ MPI supports CUDA only if the implementation used to build PyTorch supports it.
...
 are built and included in PyTorch distributed (NCCL only when building with CUDA).
 MPI is an
 optional backend that can only be included if you build PyTorch from source. (e.g.
 building PyTorch on a host that has MPI installed.)
 
+.. note ::
```
Comment (mrshenli) [inline docs/source/distributed.rst:61]: Let's use `.. warning::` instead of `.. note::`

File: docs/source/distributed.rst [hunk 5]
```diff
@@ -52,12 +52,19 @@ MPI supports CUDA only if the implementation used to build PyTorch supports it.
...
+    On Windows, Pytorch distributed currently only supports Gloo backend, and FileStore as
```
Comment (mrshenli) [inline docs/source/distributed.rst:62]:
```
As of PyTorch v1.7, Windows support for the distributed package only covers collective 
communications with Gloo backend, `FileStore`, and `DistributedDataParallel`.  Therefore,
 the `init_method` argument in :func:`init_process_group` must point to a file. This 
works for both local and shared file systems: local file system example 
``init_method="file:///d:/tmp/some_file"``, shared file system example 
``init_method="file://////{machine_name}/{share_folder_name}/some_file"``.
Similarly, if you directly pass in a `store` argument, it must be a ``FileStore``
instance. 
```

File: torch/distributed/__init__.py [hunk 6]
```diff
@@ -6,9 +6,9 @@ def is_available():
     """
     Returns ``True`` if the distributed package is available. Otherwise,
     ``torch.distributed`` does not expose any other APIs. Currently,
-    ``torch.distributed`` is available on Linux and MacOS. Set
+    ``torch.distributed`` is available on Linux, MacOS and Windows. Set
     ``USE_DISTRIBUTED=1`` to enable it when building PyTorch from source.
-    Currently, the default value is ``USE_DISTRIBUTED=1`` for Linux and
+    Currently, the default value is ``USE_DISTRIBUTED=1`` for Linux and Windows,
```
Comment (mrshenli) [inline torch/distributed/__init__.py:11]: I wonder if we should make ``USE_DISTRIBUTED=0`` by default, and let users set it to 0 when necessary?

**Review by mrshenli — APPROVED** [review]

**Pull request closed without merging**